### PR TITLE
Adding support for api-version query parameter

### PR DESCRIPTION
**Pull request by annelo-msft**
Service team has added versioning to service by adding a query string parameter.  Adding support for this in the client.

Docmentation: https://msazure.visualstudio.com/Azure%20AppConfig/_git/AppConfigService?path=%2Fdocs%2Fprotocol%2Fversioning.md&version=GBmaster

Addresses both #7502 and #6944.

### CODE REVIEW

**Comment by annelo-msft** [comment]
Tests need to be re-recorded.

File: sdk/appconfiguration/Azure.Data.AppConfiguration/src/ApiVersionPolicy.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+// Licensed under the MIT License. See License.txt in the project root for
+// license information.
+
+using System;
+using System.Threading.Tasks;
+using Azure.Core.Pipeline;
+
+namespace Azure.Data.AppConfiguration
+{
+    internal class ApiVersionPolicy : HttpPipelinePolicy
```
Comment (pakrym) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/src/ApiVersionPolicy.cs]: Use SynchronousHttpPipelinePolicy

File: sdk/appconfiguration/Azure.Data.AppConfiguration/tests/ConfigurationMockTests.cs [hunk 2]
```diff
@@ -61,7 +62,7 @@ public async Task Get()
 
             AssertRequestCommon(request);
             Assert.AreEqual(RequestMethod.Get, request.Method);
-            Assert.AreEqual("https://contoso.appconfig.io/kv/test_key", request.UriBuilder.ToString());
+            Assert.AreEqual(AppendPipelineParams("https://contoso.appconfig.io/kv/test_key"), request.UriBuilder.ToString());
```
Comment (pakrym) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/tests/ConfigurationMockTests.cs]: I would just append api-version to all the strings, it doesn't make them that much longer and makes the test less complicated.

File: sdk/appconfiguration/Azure.Data.AppConfiguration/src/ApiVersionPolicy.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    {
+        private string _versionString;
```
Comment (pakrym) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/src/ApiVersionPolicy.cs]: readonly

**Review by pakrym — APPROVED** [review]
LGTM pending comments

**Pull request merged**